components/canvas/webgl_thread.rs

```diff
@@ -273,6 +366,7 @@ impl<VR: WebVRRenderHandler + 'static> WebGLThread<VR> {
         // It is important that the fence sync is properly flushed into the GPU's command queue.
         // Without proper flushing, the sync object may never be signaled.
         data.ctx.gl().flush();
+        let gl_sync = 0;
```

Comment:
This is the cause of the `glPushDebugGroupKHR()` failure. This line should be removed.

The failure is incorrectly attributed to `glPushDebugGroupKHR()` because nothing calls `glGetError()` after `glWaitSync()`, so the failure is not detected at that point. The next check occurs after `glPushDebugGroupKHR()`.

Reply:
aieeeeeeeeee that was from when I ripped out the off-thread stuff and needed the result to compile. Thanks for spotting it!

Reply:
It is also possible that this line causes the glitching, because without a successful call to `glWaitSync()` there is no guarantee that the two contexts are properly synchronized.

Reply:
The syncing only is used in the off-thread case. Maybe it's worth adding it to the main thread implementation as well?